### PR TITLE
use Makefile similar to grpcurl, fix lint issues

### PR DESCRIPTION
**Pull request by jhump**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+updatedeps:
+	go get -d -v -t -u -f ./...
+
+.PHONY: install
+install:
+	go install ./...
+
+.PHONY: checkgofmt
+checkgofmt:
+	gofmt -s -l .
```
Comment (ianrose14) [inline Makefile]: what's the first call to `gofmt` for?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	predeclared .
+
+# Intentionally omitted from CI, but target here for ad-hoc reports.
+.PHONY: golint
+golint:
+	@go get github.com/golang/lint/golint
+	golint -min_confidence 0.9 -set_exit_status ./...
+
+# Intentionally omitted from CI, but target here for ad-hoc reports.
+.PHONY:
```
Comment (ianrose14) [inline Makefile]: meant to put `errcheck` at the end of this line?

File: Makefile (hunk 1 above)
Comment (jhump) [reply Makefile]: To actually print the violations to the test output, so the user can see what files failed the check.

File: Makefile (hunk 2 above)
Comment (jhump) [reply Makefile]: Doh! Yes.

**Pull request merged**